includes/class-analytics.php

```diff
@@ -0,0 +1,245 @@
+			$config['triggers'][ $event['id'] ] = apply_filters( 'newspack_analytics_amp_event_config', $event_config, $event );
+		}
+
+		return $config;
+	}
+
+	/**
+	 * Inject event listeners on non-AMP pages.
+	 */
+	public static function inject_non_amp_events() {
+		if ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ) {
+			return;
+		}
+
+		$sitekit_manager = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'google-site-kit' );
+		if ( ! $sitekit_manager->is_module_active( 'analytics' ) ) {
+			return;
+		}
+
+		foreach ( self::get_events() as $event ) {
+			ob_start();
+			switch ( $event['on'] ) {
+				case 'click':
+					self::output_js_click_event( $event );
+					break;
+				case 'scroll':
+					self::output_js_scroll_event( $event );
+					break;
+				default:
+					break;
+			}
+
+			// Other integrations can use this filter if they need to add a custom JS event handler.
+			$event_js = apply_filters( 'newspack_analytics_event_js', ob_get_clean(), $event );
+			echo $event_js; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		}
+	}
+
+	/**
+	 * Output JS for a click-based event listener.
+	 *
+	 * @param array $event Event info. See 'get_events'.
+	 */
+	protected static function output_js_click_event( $event ) {
+		?>
+		<script>
+			( function() {
+				var elementSelector = '<?php echo esc_attr( $event['element'] ); ?>';
+				var elements        = Array.prototype.slice.call( document.querySelectorAll( elementSelector ) );
+
+				for ( var i = 0; i < elements.length; ++i ) {
+					elements[i].addEventListener( 'click', function() {
+						gtag(
+							'event',
+							'<?php echo esc_attr( $event['event_name'] ); ?>',
+							{
+								event_category: '<?php echo esc_attr( $event['event_category'] ); ?>',
+								event_label: '<?php echo esc_attr( $event['event_label'] ); ?>'
+							}
+						);
+					} );
+				}
+			} )();
+		</script>
+		<?php
+	}
+
+	/**
+	 * Output JS for a scroll-based event listener.
+	 *
+	 * @param array $event Event info. See 'get_events'.
+	 */
+	protected static function output_js_scroll_event( $event ) {
+		?>
+		<script>
+			( function() {
+				var scrollPercent = <?php echo (int) $event['scrollSpec']['verticalBoundaries'][0]; ?>;
+
+				var eventSent = false;
+				var reportEvent = function(){
+					if ( eventSent ) {
+						return;
+					}
+
+					var scrollPos = ( window.pageYOffset || window.scrollY ) + window.innerHeight;
+					var documentHeight = document.body.clientHeight;
+
+					if ( ( ( scrollPos / documentHeight ) * 100 ) >= scrollPercent ) {
+						eventSent = true;
+						gtag(
+							'event',
+							'<?php echo esc_attr( $event['event_name'] ); ?>',
+							{
+								event_category: '<?php echo esc_attr( $event['event_category'] ); ?>',
+								event_label: '<?php echo esc_attr( $event['event_label'] ); ?>',
+								value: scrollPercent
+							}
```

Comment:
`'non_interaction': true` should be [added here](https://developers.google.com/analytics/devguides/collection/gtagjs/events#send_non-interaction_events), as we agreed to send scroll events as [non-interaction events](https://support.google.com/analytics/answer/1033068#NonInteractionEvents). One thing to check is whether the `amp-analytics` events will be sent as non-interaction events.

Reply:
Yes I had it on my list but forgot. Good catch!

Reply:
Added support in https://github.com/Automattic/newspack-plugin/pull/531/commits/c0763b1c3079771315ffe80d98dd97caa78150c0. By default events will be non-interactive with ability to mark them as interactive, which seems a safe default.